nixos/modules/services/web-apps/discourse.nix

```diff
@@ -713,11 +710,12 @@ in
                   cfg.siteSettings
                   "/run/discourse/config/nixos_site_settings.json"
               }
-              install -T -m 0400 -o discourse ${discourseConf} /run/discourse/config/discourse.conf
+              install -T -m 0600 -o discourse ${discourseConf} /run/discourse/config/discourse.conf
```

Comment:
Is this change related?

Reply:
Yes, this is required for it to be able to write to the file. `replace-literal` ignores permissions (or maybe the `-f` flag also has that effect), but `replace-secret` doesn't.